test/attr/attr_downgrade_exhaustivity.swift

```diff
@@ -0,0 +1,43 @@
+// RUN: %target-typecheck-verify-swift -swift-version 3 %s
+
+enum Runcible {
+  case spoon
+  case hat
+  @_downgrade_exhaustivity_check
+  case fork
+}
+
+enum Trioptional {
+  case some(Runcible)
+  case just(Runcible)
+  case none
+}
+
+enum Fungible {
+  case cash
+  case giftCard
+}
+
+func missingCases(x: Runcible?, y: Runcible?, z: Trioptional) {
+  // Should warn in S3 mode that `fork` isn't used
+  switch x! { // expected-warning {{switch must be exhaustive}}
+  // expected-note@-1 {{add missing case: '.fork'}}
+  case .spoon:
+    break
+  case .hat:
+    break
+  }
+
+  // Should warn in S3 mode that `fork` isn't used
+  switch (x!, y!) { // expected-warning {{switch must be exhaustive}}
+  // expected-note@-1 10 {{add missing case:}}
+  case (.spoon, .spoon):
+    break
+  case (.spoon, .hat):
+    break
+  case (.hat, .spoon):
+    break
+  case (.hat, .hat):
+    break
+  }
+}
```

Comment:
Should we raise an error if `fork` is nonexhaustively used in parts of the pattern, as in:

```
switch (x!, y!) {
case (.spoon, _):
  break
case (.hat, _):
  break
case (.fork, .spoon):
  break
case (.fork, .hat):
  break
}
```

Is it worth testing other patterns where the enum appears in structural position, like nested in `Optionals` or `Trioptional` declared above?